plugins/reboot/reboot.go

```diff
@@ -50,24 +102,51 @@ func (plugin) NewTaskPlugin(options plugins.TaskPluginOptions) (plugins.TaskPlug
 		return nil, engines.ErrContractViolation
 	}
 
+	pl.runningTasks.RLock()
 	return taskPlugin{
 		TaskPluginBase: plugins.TaskPluginBase{},
-		reboot:         p.Reboot,
+		plugin:         pl,
+		rebootWhenDone: p.RebootWhenDone,
 	}, nil
 }
 
 func (tp taskPlugin) Dispose() error {
```

Comment:
I was curious why it was passed by value.  All the work I've been doing in Rust made me think the compiler would take care of alerting me if I was copying some value that shouldn't be copied (like, in this case, the RWLock).